cmd/gogio/jsbuild.go

```diff
@@ -39,9 +40,41 @@ func buildJS(bi *buildInfo) error {
 	if err != nil {
 		return err
 	}
-	if err := ioutil.WriteFile(filepath.Join(out, "index.html"), []byte(jsIndex), 0600); err != nil {
+
+	var faviconPath string
+	if _, err := os.Stat(bi.iconPath); err == nil {
+		// Copy icon to the output folder
+		icon, err := ioutil.ReadFile(bi.iconPath)
+		if err != nil {
+			return err
+		}
+		if err := ioutil.WriteFile(filepath.Join(out, filepath.Base(bi.iconPath)), icon, 600); err != nil {
+			return err
+		}
+		faviconPath = filepath.Base(bi.iconPath)
+	}
+
+	indexTemplate, err := template.New("").Parse(jsIndex)
+	if err != nil {
 		return err
 	}
+
+	index, err := os.Create(filepath.Join(out, "index.html"))
+	if err != nil {
+		return err
+	}
+	defer index.Close()
```

Comment:
A failing Close may leave the icon corrupted. Either check the return value or buffer `index.html` and use WriteFile.